tests/unit/ChatPopup.spec.ts

```diff
@@ -0,0 +1,86 @@
+import axios from 'axios'
+import sinon from 'sinon'
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import Vuex from 'vuex'
+import { mount, createLocalVue } from '@vue/test-utils'
+import ChatPopup from '@/components/common/ChatPopup.vue'
+
+// Prevent the warning "[Vuetify] Unable to locate target [data-app]"
+document.body.setAttribute('data-app', 'true')
+
+Vue.use(Vuetify)
+Vue.use(Vuex)
+
+const localVue = createLocalVue()
+const vuetify = new Vuetify({})
+
+// stub isMobile, which we can't call during testing
+const store = new Vuex.Store<any>({
+  getters: {
+    isMobile: () => false
+  }
+})
+
+describe('ChatPopup component', () => {
+  let wrapper: any
+
+  beforeEach(async () => {
+    window['webChatReason'] = 'Test'
+    window['webChatStatusUrl'] = 'myhost/basePath/webchatStatus'
+    window['webChatUrl'] = 'myhost/basePath/webchatUrl'
+
+    // stub GET webchat status
+    sinon.stub(axios, 'get').withArgs('myhost/basePath/webchatStatus')
+      .returns(new Promise((resolve) => resolve({
+        data: { 'status': 'open' }
+      })))
+
+    wrapper = mount(ChatPopup, { vuetify, localVue, store })
+    await Vue.nextTick()
+  })
+
+  afterEach(() => {
+    sinon.restore()
+    wrapper.destroy()
+  })
+
+  it('renders the components', () => {
+    expect(wrapper.find(ChatPopup).exists()).toBe(true)
+    expect(wrapper.find('#chat-button-wrapper').exists()).toBe(true)
+    expect(wrapper.find('#open-tooltip-message').exists()).toBe(false)
+    expect(wrapper.find('#closed-tooltip-message').exists()).toBe(false)
+    expect(wrapper.find('#unavailable-tooltip-message').exists()).toBe(false)
+  })
+
+  it('displays the tooltip for the open webchat status', async (done) => {
```

Comment:
What is the ```done``` doing in the ```async (done)```?

Reply:
The test waits for this function to be called when the test code is complete (otherwise the test times out and fails).

This is how async code previously worked, ie., promises. Now, in most cases -- but not here -- we use inline `await` for async execution.

Reply:
@lambert-alex Please resolve this conversation if OK (since you started it).